Additional ecdsa tests prototype

chipshort opened the pull request.

aumetra commented on packages/crypto/tests/rootberg_secp256k1.rs:
```diff
@@ -0,0 +1,164 @@
+use cosmwasm_crypto::{secp256k1_recover_pubkey, secp256k1_verify};
+use serde::Deserialize;
+
+// See ./testdata/wycheproof/README.md for how to get/update those files
```
This still references the old path

```suggestion
// See ./testdata/rootberg/README.md for how to get/update those files
```

aumetra commented on packages/crypto/tests/rootberg_secp256k1.rs:
```diff
@@ -0,0 +1,164 @@
+        .s
+        .iter()
+        .enumerate()
+        .find(|(_, v)| **v != 0)
```
This is just a reimplementation of [`.position()`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.position)

```suggestion
        .position(|item| *v != 0)
```

chipshort replied:
Oh, that's much nicer!

aumetra commented on packages/crypto/tests/rootberg_secp256k1.rs:
```diff
@@ -0,0 +1,164 @@
+            assert_eq!(recovered, test.pubkey);
+        }
+    }
+    assert_eq!(tested, num_tests);
```
Do we need to keep track manually here, or could we replace this with an assertion at the very top (after reading the test vectors) that looks something like this:

```rust
assert_eq!(num_tests, tests.len());
```

Since this check seems to be a sanity check whether the `num_tests` value was actually correct.

aumetra reviewed (changes requested):
A few small nits

chipshort commented:
I have updated the tests to use the non-normalized test files, which allows us to remove the ugly comment check.
I also added the r1 test files and refactored the code a bit, so we can reuse the hashers and json types.

webmaster128 commented:
Is this ready for review? I added it to the 2.1 milestone as it gives us extra confidence in the newly added r1 APIs.

chipshort commented:
Yes, the code itself is ready for review. Only thing I don't know is how stable the rootberg format is, but we can always adjust to that later if it comes up.

aumetra commented on packages/crypto/tests/rootberg_secp256r1.rs:
```diff
@@ -0,0 +1,65 @@
+    for test in tests {
+        assert_eq!(test.public_key_uncompressed.len(), 65);
+
+        // eprintln!("Test case ID: {}", test.tc_id);
```
Would it hurt to keep the `println!`? If a change to our code ever breaks the test vectors, it would be immediately obvious what broke.

And if it's not important, shouldn't we remove this then?

aumetra commented on packages/crypto/tests/rootberg_secp256k1.rs:
```diff
@@ -0,0 +1,65 @@
+    for test in tests {
+        assert_eq!(test.public_key_uncompressed.len(), 65);
+
+        // eprintln!("Test case ID: {}", test.tc_id);
```
Same here
